### PR TITLE
DOCS-1312: Document `MLModel` service API

### PR DESCRIPTION
**Pull request by sguequierre**
* Documents infer, metadata, and do_command for both SDKS

Needs SME verification on the input tensor creation, tried to make like tests

### CODE REVIEW

**Review by npentrel — APPROVED** [review]

File: docs/services/ml/_index.md [hunk 1]
```diff
@@ -147,6 +147,177 @@ In the absence of metadata, your `.tflite_cpu` model must satisfy the following
...
+
+For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/viam/services/mlmodel/client/index.html#viam.services.mlmodel.client.MLModelClient.infer).
+
+```python {class="line-numbers linkable-line-numbers"}
+my_mlmodel = MLModelClient.from_robot(robot=robot, name="my_mlmodel_service")
+
+nd_array = np.array([1, 2, 3], dtype=np.float64)
+input_tensors = {"0": nd_array}
+
+output_tensors = await my_mlmodel.infer(input_tensors)
```
Comment (hexbabe) [inline docs/services/ml/_index.md:210]: Have we confirmed that this works— sry I don't think I've written client code for ml model

File: docs/services/ml/_index.md [hunk 2]
```diff
@@ -147,6 +147,177 @@ In the absence of metadata, your `.tflite_cpu` model must satisfy the following
...
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+### Metadata
+
+Get the metadata: name, type, expected tensor/array shape, inputs, and outputs associated with the ML model.
```
Comment (hexbabe) [inline docs/services/ml/_index.md:218]: Maybe `data type` instead of `type`?

Comment (sguequierre) [inline docs/services/ml/_index.md:218]:
```suggestion
Get the metadata: name, data type, expected tensor/array shape, inputs, and outputs associated with the ML model.
```

File: docs/services/ml/_index.md (hunk 1 above)
Comment (sguequierre) [inline docs/services/ml/_index.md:210]: I tried to test but I think that I have to have a service with a model in it that expects that exact input shape and I'm not sure how to do that. @cheukt Does this look like it should be ok as a basic example, or is there something that I could do that would work better to demo?

File: docs/services/ml/_index.md [hunk 3]
```diff
@@ -147,6 +147,177 @@ In the absence of metadata, your `.tflite_cpu` model must satisfy the following
...
+- `timeout` [(Optional\[float\])](https://docs.python.org/library/typing.html#typing.Optional): An option to set how long to wait (in seconds) before calling a time-out and closing the underlying RPC call.
+
+**Returns:**
+
+- [(`Dict[str, NDArray]`)](https://numpy.org/doc/stable/reference/generated/numpy.ndarray.html): A dictionary of output flat tensors as specified in the metadata, after being run through an inference engine.
+
+For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/viam/services/mlmodel/client/index.html#viam.services.mlmodel.client.MLModelClient.infer).
+
+```python {class="line-numbers linkable-line-numbers"}
+my_mlmodel = MLModelClient.from_robot(robot=robot, name="my_mlmodel_service")
```
Comment (sguequierre) [inline docs/services/ml/_index.md:205]:
```suggestion
import numpy as np

my_mlmodel = MLModelClient.from_robot(robot=robot, name="my_mlmodel_service")
```

File: docs/services/ml/_index.md (hunk 1 above)
Comment (cheukt) [reply docs/services/ml/_index.md:212]: I think that should work. the tests I linked above should be a good example

File: docs/services/ml/_index.md (hunk 2 above)
Comment (sguequierre) [reply docs/services/ml/_index.md]: Done

File: docs/services/ml/_index.md (hunk 1 above)
Comment (sguequierre) [reply docs/services/ml/_index.md:212]: Ok, thanks!

**Review by cheukt — APPROVED** [review]

**Pull request merged**